javaagent-internal-logging-application/src/main/java/io/opentelemetry/javaagent/logging/application/InMemoryLog.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.logging.application;
+
+import io.opentelemetry.javaagent.bootstrap.InternalLogger;
+import java.io.PrintStream;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+final class InMemoryLog {
```

Comment:
do we have AutoValue around here? Might it help to take away some of the boilerplate?